### PR TITLE
perf(utils): add benchmarks for file system walk and glob libs

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR includes perf benchmarks for:
- file system crawling
  - [@nodelib/fs.walk](https://www.npmjs.com/package/@nodelib/fs.walk)
- glob search libs
  - [fast-glob](https://www.npmjs.com/package/fast-glob)
  - [glob](https://www.npmjs.com/package/glob)
  - [globby](https://www.npmjs.com/package/globby)

---

**file system crawling**  

```bash
> nx run utils:perf crawl-file-system

You can adjust the test with the following arguments:
directory      target directory of test      --directory=/Users/michael_hladky/WebstormProjects/quality-metrics-cli/node_modules
pattern        pattern to search             --pattern=/.json$/
 
Start benchmark...
 
Base x 2.43 ops/sec ±1.45% (17 runs sampled)
nodelib.fsWalk x 17.09 ops/sec ±0.74% (82 runs sampled)
 
Total Duration: 13.08 sec
Fastest is nodelib.fsWalk
```

**glob search libs**  
```bash
> nx run utils:perf glob

You can adjust the test with the following arguments:
pattern      glob pattern of test      --pattern=/Users/michael_hladky/WebstormProjects/quality-metrics-cli/node_modules/**/*.js
 
Start benchmark...
 
glob found 27167 files
glob x 11.66 ops/sec ±19.61% (67 runs sampled)
globby found 27164 files
globby x 9.18 ops/sec ±0.74% (48 runs sampled)
fastGlob found 27164 files
fastGlob x 12.62 ops/sec ±0.76% (63 runs sampled)
 
Total Duration: 18.21 sec
Fastest is fastGlob
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -128,7 +128,11 @@
     "verdaccio": "^5.0.4",
     "vite": "~4.3.9",
     "vitest": "~0.32.0",
-    "zod2md": "^0.1.1"
+    "zod2md": "^0.1.1",
+    "@nodelib/fs.walk": "^2.0.0",
+    "fast-glob": "^3.3.2",
+    "glob": "^10.3.10",
+    "globby": "^14.0.1"
```
Comment (matejchalk) [inline package.json:135]: Could you sort these, please? A trick I use is to do something like `npm rm savbjhsfbvadsfbv` (just type anything that's not a real package name :grin:), it has the side-effect of sorting it automatically.

**Review by matejchalk — COMMENTED** [review]
Cool, so `fast-glob` is correctly named :grin:

File: package.json (hunk 1 above)
Comment (BioPhoton) [reply package.json]: done!

Comment (BioPhoton) [reply package.json]: BTW, could we intro `npm rm non-existin-lib` in or format task?

Comment (matejchalk) [reply package.json]: There is a [Prettier plugin for `package.json`](https://github.com/matzkoh/prettier-plugin-packagejson) :bulb:

**Pull request merged**